### PR TITLE
Escapes commas in displayName when exporting users in CSV format.

### PR DESCRIPTION
**Pull request by bojeil-google**
Escapes commas in displayName when exporting users in CSV format.
This will encapsulate the string with double quotes.

### CODE REVIEW

File: src/test/accountExporter.spec.js [hunk 1]
```diff
@@ -113,5 +114,54 @@ describe("accountExporter", function() {
...
+          }
+          var expectedEntry =
+            userList[j].localId +
+            "," +
+            userList[j].email +
+            ",false,,," +
+            expectedDisplayName +
+            trailingCommas.join("");
+          expect(spyWrite.getCall(j).args[0]).to.eq(expectedEntry + "," + os.EOL);
+        }
```
Comment (bkendall) [inline src/test/accountExporter.spec.js]: So if the `userList` above is adjusted to be a single user with a comma in their name, do we need this loop?

Further, do we need to check this string exactly? it seems like that is done (in a complex fashion) above in the other test.

This test could probably be asserted by saying the following:

```javascript
const writeArg = spyWrite.getCall(j).args[0];
expect(writeArg).to.contain(`,"Display Name, with Comma",`);
```

Re-testing that the entire line is correct again is _probably_ overkill in this situation - then, you're testing that a multitude of things are working, instead of just one thing.

File: src/accountExporter.js [hunk 2]
```diff
@@ -30,14 +30,22 @@ var PROVIDER_ID_INDEX_MAP = {
   "github.com": 19,
 };
 
+var _escapeComma = function(str) {
+  if (str.indexOf(",") !== -1) {
+    // Encapsulate the string with quotes if it contains a comma.
+    return '"' + str + '"';
```
[the remark this answers is not in the document]
Comment (bojeil-google) [reply src/accountExporter.js]: Done.

File: src/test/accountExporter.spec.js [hunk 3]
```diff
@@ -50,6 +50,7 @@ describe("accountExporter", function() {
         userList.push({
           localId: i.toString(),
           email: "test" + i + "@test.org",
+          displayName: "John Tester" + i + (i === 0 ? ", CFA" : ""),
```
[the remark this answers is not in the document]
Comment (bojeil-google) [reply src/test/accountExporter.spec.js]: Thanks for catching.
I can either hardcode the user list with one comma case or write a conditional. I initially opted for the latter.  I would like to also test csv behavior in general. Current test coverage is extremely low (there was no test case for csv flow). So I added a test for general csv behavior and kept the test for just the comma in display name edge case (using a single user).

File: src/test/accountExporter.spec.js [hunk 4]
```diff
@@ -113,5 +114,54 @@ describe("accountExporter", function() {
...
+          nextPageToken: "7",
+        });
+
+      var result = serialExportUsers("test-project-id", {
+        format: "csv",
+        batchSize: 7,
+        writeStream: writeStream,
+      });
+      return result.then(function() {
+        expect(spyWrite.callCount).to.eq(7);
```
[the remark this answers is not in the document]
Comment (bojeil-google) [reply src/test/accountExporter.spec.js]: Updated to use `userList.length`.

File: src/test/accountExporter.spec.js (hunk 1 above)
Comment (bojeil-google) [reply src/test/accountExporter.spec.js]: Added one test for csv behavior and updated this test to just test comma encapsulation on single user.

File: src/test/accountExporter.spec.js [hunk 5]
```diff
@@ -113,5 +114,54 @@ describe("accountExporter", function() {
...
+      for (var i = 0; i < 20; i++) {
+        trailingCommas.push(",");
+      }
+      nock("https://www.googleapis.com")
+        .post("/identitytoolkit/v3/relyingparty/downloadAccount", {
+          maxResults: 7,
+          targetProjectId: "test-project-id",
+        })
+        .reply(200, {
+          users: userList,
```
[the remark this answers is not in the document]
Comment (bojeil-google) [reply src/test/accountExporter.spec.js]: I made the change. Check above for more details.

File: src/test/accountExporter.spec.js [hunk 6]
```diff
@@ -113,5 +114,119 @@ describe("accountExporter", function() {
...
+          maxResults: 1,
+          nextPageToken: "1",
+          targetProjectId: "test-project-id",
+        })
+        .reply(200, {
+          users: [],
+          nextPageToken: "1",
+        });
+
+      var result = serialExportUsers("test-project-id", {
```
Comment (bkendall) [inline src/test/accountExporter.spec.js]: These are looking pretty good. I think the last adjustment that should be made is `result` here - it's really not being used. `serialExportUsers` returns a promise, can this return the promise chain starting there?

```javascript
return serialExportUsers("...")
  .then(function () {
    expect...
  });

Comment (bojeil-google) [reply src/test/accountExporter.spec.js]: Done.
